Pasteing of comma and space separated lists

bmordan opened the pull request:
This PR enables uses to paste a list of either:
`'comma','separates','values'`
or
`'words' 'separated' 'with' spaces`
and
have **react-tag-input** split the values up and add them one by one to the field.

Tidy 📋

bmordan committed:
enables pasting of comma and space separated lists

alanshaw commented:
👓

alanshaw commented on dist-modules/reactTags.js:
```diff
@@ -179,6 +178,20 @@ var ReactTags = React.createClass({
+
+        if (commas.length > spaces.length) commas.forEach(function (buzzword) {
+            return _this.props.handleAddition(buzzword);
+        });else spaces.forEach(function (buzzword) {
```
Syntax error here

alanshaw commented on dist-modules/reactTags.js:
```diff
@@ -179,6 +178,20 @@ var ReactTags = React.createClass({
+        e.preventDefault();
+        var string = e.clipboardData.getData('Text');
+        var commas = string.split(',');
+        var spaces = string.split(' ');
```
Split on space or comma?

``` js
string.split(/[ ,]+/)
```

``` js
$ node
> 'and then, there,were,  some ,   words'.split(/[ ,]+/)
[ 'and', 'then', 'there', 'were', 'some', 'words' ]
```

alanshaw commented:
You didn't make all the changes you made in `lib/reactTags.js` as you made in `dist-modules/reactTags.js`. I think the latter is generated anyway so you should probably use the build process for doing that if there is one.

bmordan committed:
refactor after Alans review

alanshaw commented on lib/reactTags.js:
```diff
@@ -168,6 +167,11 @@ var ReactTags = React.createClass({
     },
+    handlePaste: function (e) {
+      e.preventDefault()
+      let string = e.clipboardData.getData('Text')
```
In IE11 this causes an error. We need to check for `e.clipboardData` and if it doesn't exist get it from `window` (where IE11 keeps it).

``` js
const clipboardData = e.clipboardData || window.clipboardData
const string = clipboardData.getData('text')
```

Also note that lowercase "text" seems to be the standard.

bmordan committed:
catch for EI11 clipboard data. removes generated files from repo

bmordan commented:
I have removed and `.gitignore` the `/dist` and `dist-modules` folders.
I added a `postinstall` step to generate these files.
There is no need to persist these in the repo, and they are a bit confusing when looking at the diffing on a PR

alanshaw commented:
In future please send unrelated changes as a separate PR
